firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexBackfiller.java

```diff
@@ -169,8 +174,15 @@ private int writeEntriesForCollectionGroup(
     ImmutableSortedMap<DocumentKey, Document> documents =
         localDocumentsView.getDocumentsMatchingQuery(query, earliestUpdateTime);
 
-    Queue<Document> oldestDocuments = getOldestDocuments(documents, entriesRemainingUnderCap);
+    Queue<Document> oldestDocuments = getOldestDocuments(documents, documentsRemaining);
     indexManager.updateIndexEntries(oldestDocuments);
+
+    // Mark the collection group as fully indexed if all documents in the collection have been
+    // indexed during this backfill iteration.
+    if (documentsRemaining >= documents.size()) {
+      indexManager.markCollectionGroupIndexed(collectionGroup);
```

Comment:
This is different from how I implemented it. I always increment the sequence number regardless of whether I am done or not - it's a tiny bit simpler but may not be as optimal as what you are doing. We should discuss.